.travis.yml

```diff
@@ -1,55 +1,114 @@
   install: false
+  before_script: false
   script: false
+  after_success: false
+  after_failure: false
+  before_deploy: false
+  deploy: false
 
-.check_moban: &check_moban
+.moban: &moban
   <<: *disable_global
   python: 3.6
   stage: moban
-  install: pip install moban
-  script: .ci/check_moban.sh
+  install: pip install moban>=0.0.4
+  script:
+    - moban
+    - git diff --exit-code
 
 jobs:
   include:
-    - *check_moban
+    - stage: moban
+      <<: *moban
+      if: branch = master AND type = push
+    - *moban
+  allow_failures:
+    - *moban
+
+stage: test
+
+env:
+  global:
+    - BEARS_ZIP_URL=https://codeload.github.com/coala/coala-bears/zip
+    - GRAVIS="https://raw.githubusercontent.com/DanySK/Gravis-CI/master/"
+    - JDK="adopt@1.8.212-04"
 
 cache:
   pip: true
   directories:
-    # coala managed data
     - ~/nltk_data
+    - ~/.jabba
 
 before_install:
-  - pip install setuptools -U
-  - printf '%s\n'
-           "$(cat test-requirements.txt requirements.txt)"
-           > requirements.txt
+  - curl "${GRAVIS}.install-jdk-travis.sh" --output ~/.install-jdk-travis.sh
 
-script:
-  - pytest
-  - pip install -e .
-  # https://github.com/coala/coala-bears/issues/1037
   - >
-    if [[ "$TRAVIS_PULL_REQUEST" != "false" ]]; then
-      sed -i.bak '/bears = GitCommitBear/d' .coafile
+    if [[ $TRAVIS_OS_NAME == osx ]]; then
+      brew install python3
+      sudo ln -sf $(which python3) $(which python)
+      sudo ln -sf $(which pip3) $(which pip 2>/dev/null || echo /usr/bin/pip)
     fi
+
+  # See rationale in .ci/deps.python-packages.ps1 for pre-installing these
+  - pip install --prefer-binary cffi lxml
+  # Use pip 9, so that setuptools can be downgraded.
+  # https://github.com/coala/coala/issues/4708
+  # Apart from builds with python 2.7 and 3.5, it installs setuptools
+  # twice. So, it is required to uninstall one version manually.
+  - pip uninstall setuptools --yes
+  - pip uninstall setuptools --yes || true
+  - pip uninstall setuptools --yes || true
+  - pip uninstall pipenv --yes || true
+  - pip install pip==9.0.1 setuptools==21.2.2
+
+  # https://github.com/coala/coala/issues/3183
+  # Travis automatically installs the `requirements.txt` in "install" stage
+  - cp requirements.txt requirements.orig
+  - cat test-requirements.txt >> requirements.txt
+  - sed -i.bak '/^-r/d' requirements.txt
+
+  # OSX is not a real Travis 'python' image, so it does not install
+  # requirements.txt automatically.
+  - if [[ $TRAVIS_OS_NAME == osx ]]; then pip install -r requirements.txt; fi
+  - source ~/.install-jdk-travis.sh
+
+before_script:
+  # Restore the original requirements.txt
+  - mv requirements.orig requirements.txt
+
+script:
+  - py.test
+  - python setup.py bdist_wheel
+  - pip install ./dist/*.whl
+  - curl -fsSL -o coala-bears.zip $BEARS_ZIP_URL/master
+  - pip install coala-bears.zip[alldeps]
   - coala --non-interactive
 
-notifications:
-  email: false
-  webhooks: https://www.travisbuddy.com/
```

Comment:
The loss of travisbuddy is the only regression here that I am aware of.

Im not sure it is worth keeping.  It has some benefits, but there hasnt been much traction on the issues I created , https://github.com/bluzi/travis-buddy/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+author%3Ajayvdb especially https://github.com/bluzi/travis-buddy/issues/117 which is very annoying as we have sometimes flaky jobs that need restarting. ping @bluzi